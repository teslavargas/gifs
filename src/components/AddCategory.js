import React, { useState } from 'react'
import PropTypes from 'prop-types'; 

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); 
    const handdleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handdleSubmit = (e) => {
        e.preventDefault(); 
        
        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue(''); 
        }
    
    }

    return (
        <form onSubmit={handdleSubmit}>
            <p> { inputValue } </p>
            <input type="text" value={inputValue} onChange={handdleInputChange}/>
        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}