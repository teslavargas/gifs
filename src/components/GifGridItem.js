import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__rubberBand">
            <img src={url} alt={title}/>
            <p className="titulo">{title}</p> 
        </div>
    )
}

GifGridItem.protoTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}