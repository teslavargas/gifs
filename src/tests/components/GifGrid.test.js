import React from 'react';
import '@testing-library/jest-dom'; 
import {GifGrid} from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
 
describe('Tests for GifGrid component', () => {
    const category = 'Goku';
    // const wrapper = shallow(<GifGrid category={category}/>);
 
    test('should show the correct component and match with the snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot(); 
    });
 
    test('should change the snapshot when useFetchGifs is executed', () => {
        
        const gifs = [{
            id: 'ABC',
            url: 'https://google.com',
            title: 'ABC Title'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false); 
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length); 
    });
})

