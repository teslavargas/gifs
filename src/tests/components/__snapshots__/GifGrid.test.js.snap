// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifGrid component should change the snapshot when useFetchGifs is executed 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInRight"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="ABC Title"
      url="https://google.com"
    />
  </div>
</Fragment>
`;

exports[`Tests for GifGrid component should show the correct component and match with the snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounceInRight"
  >
    Goku
  </h3>
  <p
    className="loading animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
